extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEmitter.java

```diff
@@ -0,0 +1,180 @@
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+
+public class GraphiteEmitter implements Emitter
+{
+  private static Logger log = new Logger(GraphiteEmitter.class);
+
+  private final GraphiteEventConverter graphiteEventConverter;
+  private final GraphiteEmitterConfig graphiteEmitterConfig;
+  private final AtomicBoolean started = new AtomicBoolean(false);
+  private final ObjectMapper mapper;
+  private final LinkedBlockingQueue<GraphiteEvent> eventsQueue;
+  private final static long DEFAULT_PUT_GET_TIMEOUT = 1000; // default wait for put/get operations on the queue 1 sec
+  private static final long FLUSH_TIMEOUT = 60000; // default flush wait 1 min
+  private final ScheduledExecutorService exec = Executors.newScheduledThreadPool(2, new ThreadFactoryBuilder()
+      .setDaemon(true)
+      .setNameFormat("GraphiteEmitter-%s")
+      .build()); // Thread pool of two in order to schedule flush runnable
+
+  public GraphiteEmitter(
+      @NotNull GraphiteEmitterConfig graphiteEmitterConfig,
+      GraphiteEventConverter graphiteEventConverter,
+      @NotNull ObjectMapper mapper
+  )
+  {
+    this.mapper = mapper;
+    this.graphiteEmitterConfig = graphiteEmitterConfig;
+    this.graphiteEventConverter = graphiteEventConverter;
+    this.eventsQueue = new LinkedBlockingQueue(graphiteEmitterConfig.getMaxQueueSize());
+  }
+
+  @Override
+  public void start()
+  {
+    log.info("Starting Graphite Emitter.");
+    synchronized (started) {
+      if (!started.get()) {
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        started.set(true);
+      }
+    }
+  }
+
+
+  @Override
+  public void emit(Event event)
+  {
+    if (!started.get()) {
+      throw new ISE("WTF emit was called while service is not started yet");
+    }
+
+    if (event instanceof ServiceMetricEvent) {
+      final ServiceMetricEvent serviceMetricEvent = (ServiceMetricEvent) event;
+      final GraphiteEvent graphiteEvent = graphiteEventConverter.druidMetricToGraphite(serviceMetricEvent);
+      if (graphiteEvent == null) {
+        return;
+      }
+      try {
+        final boolean isSuccessful = eventsQueue.offer(graphiteEvent, DEFAULT_PUT_GET_TIMEOUT, TimeUnit.MILLISECONDS);
+        if (!isSuccessful) {
+          log.error(
+              "Throwing event [%s] on the floor Graphite queue is full please increase the capacity or/and the consumer frequency",
+              mapper.writeValueAsString(event)
+          );
+        }
+      }
+      catch (InterruptedException | JsonProcessingException e) {
+        Thread.interrupted();
```

Comment:
combining JsonProcessingException with InterruptedException is probably not right, you don't want to interrupt the thread if it wansn't an InterruptedException